### PR TITLE
Callbacks for widget mount/unmount

### PR DESCRIPTION
**Pull request by VojtaSim**

### CODE REVIEW

File: packages/integration-react/src/MerkurComponent.jsx [hunk 1]
```diff
@@ -191,26 +174,40 @@ export default class MerkurComponent extends React.Component {
...
-        });
-      } catch (_) {
-        if (this.props.debug) {
-          console.warn(_);
-        }
+    const merkur = getMerkur();
+
+    try {
+      this._widget = await merkur.create(this.props.widgetProperties);
+      this._widget.mount();
```
Comment (mjancarik) [inline packages/integration-react/src/MerkurComponent.jsx:194]: Widget's mount method is async.

File: packages/integration-react/src/MerkurComponent.jsx [hunk 2]
```diff
@@ -191,26 +174,40 @@ export default class MerkurComponent extends React.Component {
...
+
+      if (typeof this.props.onWidgetMounted === 'function') {
+        this.props.onWidgetMounted();
```
Comment (mjancarik) [inline packages/integration-react/src/MerkurComponent.jsx:197]: I think that the onWidgetMounted method should get widget as argument.

File: packages/integration-react/src/MerkurComponent.jsx [hunk 3]
```diff
@@ -191,26 +174,40 @@ export default class MerkurComponent extends React.Component {
 
     this._widget.unmount();
     this._widget = null;
+
+    if (this.props.onWidgetUnmounted) {
```
Comment (mjancarik) [inline packages/integration-react/src/MerkurComponent.jsx:178]: I think that the onWidgetUnmounted method should get widget as argument and calling this function should be before calling widget.unmount. Have you got some examples where you need calling onWidgetUnmounted method after widget.unmount?

File: packages/integration-react/src/MerkurComponent.jsx (hunk 1 above)
Comment (VojtaSim) [reply packages/integration-react/src/MerkurComponent.jsx]: Added `await`

File: packages/integration-react/src/MerkurComponent.jsx (hunk 2 above)
Comment (VojtaSim) [reply packages/integration-react/src/MerkurComponent.jsx]: Passing `this._widget` to the function call

File: packages/integration-react/src/MerkurComponent.jsx (hunk 3 above)
Comment (VojtaSim) [reply packages/integration-react/src/MerkurComponent.jsx]: `onWidgetUnmounted` has been renamed to `onWidgetUnmounting` and is called before the actual unmount call. Also it receives `this._widget` as an argument.

**Pull request merged**